src/logger.cpp

```diff
@@ -0,0 +1,103 @@
+#include "logger.hpp"
+
+thread_local Logger::ThreadLocals Logger::tl;
+
+Logger& GetLogger(const int _linenr, const char* _filename, const char* _function)
+{
+  static Logger log;
+  log.setLocation(_linenr, _filename, _function);
+  return log;
+}
+
+void Logger::writelog(const LogLevel _level, const std::string& _line, const Location _loc)
+{
+  if (level2log < _level) {
+    return;
+  }
+  std::ostringstream line;
+
+  std::array<char, 50> datetime{};
+  datetime[0] = '\0';
+
+  struct tm tm;
+  time_t t;
+  time(&t);
+  localtime_r(&t, &tm);
+  if (strftime(datetime.data(), datetime.size(), "%Y-%m-%d %H:%M:%S ", &tm) == 0) {
+    datetime[0] = '\0';
+  }
+
+  line << datetime.data();
+  std::map<LogLevel, std::string> loglevel2line = {
```

Comment:
```suggestion
  static std::map<LogLevel, std::string> loglevel2line = {
```

should avoid making a new map every time, I think